File: backend/src/Domain/Entities/Station.cs
using Domain.Common;
using Domain.Enums;
using Domain.Users;

namespace Domain.Entities;

public class Station : AuditableEntity
{
    private Station() { }

    private Station(string name, string address, double latitude, double longitude, Guid ownerId)
    {
        Name = name;
        Address = address;
        Latitude = latitude;
        Longitude = longitude;
        OwnerId = ownerId;
        Status = StationStatus.Pending;
        Chargers = new List<Charger>();
        OperatingHours = new List<OperatingHour>();
    }

    public Guid Id { get; private set; }
    public string Name { get; private set; } = null!;
    public string Address { get; private set; } = null!;
    public double Latitude { get; private set; }
    public double Longitude { get; private set; }
    public Guid OwnerId { get; private set; }
    
    public StationStatus Status { get; private set; }
    public string? RejectionReason { get; private set; }

    public ICollection<Charger> Chargers { get; private set; } = null!;
    public ICollection<OperatingHour> OperatingHours { get; private set; } = null!;

    public User Owner { get; private set; } = null!;

    public static Station Create(string name, string address, double latitude, double longitude, Guid ownerId)
    {
        if (string.IsNullOrWhiteSpace(name))
            throw new ArgumentException("Station name is required.", nameof(name));
        if (string.IsNullOrWhiteSpace(address))
            throw new ArgumentException("Station address is required.", nameof(address));

        return new Station(name.Trim(), address.Trim(), latitude, longitude, ownerId);
    }

    public void Approve()
    {
        Status = StationStatus.Active;
        RejectionReason = null;
    }

    public void Reject(string reason)
    {
        if (string.IsNullOrWhiteSpace(reason))
            throw new ArgumentException("Rejection reason is required.", nameof(reason));

        Status = StationStatus.Rejected;
        RejectionReason = reason;
    }
    
    public void Suspend()
    {
        Status = StationStatus.Suspended;
    }

    public void UpdateDetails(string name, string address, double latitude, double longitude)
    {
        if (string.IsNullOrWhiteSpace(name))
            throw new ArgumentException("Station name is required.", nameof(name));
        if (string.IsNullOrWhiteSpace(address))
            throw new ArgumentException("Station address is required.", nameof(address));

        Name = name.Trim();
        Address = address.Trim();
        Latitude = latitude;
        Longitude = longitude;
    }

    public void AddCharger(Charger charger)
    {
        Chargers.Add(charger);
    }
    
    public void UpdateOperatingHours(List<OperatingHour> newHours)
    {
        OperatingHours.Clear();
        foreach (var oh in newHours)
        {
            OperatingHours.Add(oh);
        }
    }
}

File: mobile-flutter/lib/features/stations/models/station.dart
import 'charger.dart';

class Station {
  final String id;
  final String name;
  final String address;
  final double latitude;
  final double longitude;
  final String ownerId;
  final String status;
  final String? rejectionReason;
  final List<Charger>? chargers;

  Station({
    required this.id,
    required this.name,
    required this.address,
    required this.latitude,
    required this.longitude,
    required this.ownerId,
    required this.status,
    this.rejectionReason,
    this.chargers,
  });

  factory Station.fromJson(Map<String, dynamic> json) {
    return Station(
      id: json['id'] as String,
      name: json['name'] as String,
      address: json['address'] as String,
      latitude: (json['latitude'] as num).toDouble(),
      longitude: (json['longitude'] as num).toDouble(),
      ownerId: json['ownerId'] as String,
      status: json['status'] as String,
      rejectionReason: json['rejectionReason'] as String?,
      chargers: json['chargers'] != null
          ? (json['chargers'] as List)
              .map((e) => Charger.fromJson(e as Map<String, dynamic>))
              .toList()
          : null
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'address': address,
      'latitude': latitude,
      'longitude': longitude,
      'ownerId': ownerId,
      'status': status,
      'rejectionReason': rejectionReason,
      if (chargers != null) 'chargers': chargers!.map((e) => e.toJson()).toList(),
    };
  }
}
